src/transformStateWithClones.js

```diff
@@ -7,7 +7,38 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let curentState = { ...state };
+  const stateArr = [];
+
+  for (let i = 0; i < actions.length; i++) {
+    const action = actions[i];
+    const st = { ...curentState };
+
+    switch (action.type) {
+      case 'addProperties':
+        for (const key in action.extraData) {
+          st[key] = action.extraData[key];
+        }
+        break;
+      case 'removeProperties':
+        for (let j = 0; j < action.keysToRemove.length; j++) {
+          const key = action.keysToRemove[j];
+
+          delete st[key];
+        }
+        break;
+      case 'clear':
+        for (const key in st) {
+          delete st[key];
+        }
+        break;
+    }
```

Comment:
Consider adding a default case in the switch statement to handle unexpected action types. This will help in debugging and ensure that the function behaves predictably when an unknown action type is encountered.